src/OpenLoco/src/World/CompanyAi/CompanyAi.cpp

```diff
@@ -2155,12 +2155,197 @@ namespace OpenLoco
+        {
+            return true;
+        }
+
+        if ((company.aiPlaystyleFlags & AiPlaystyleFlags::unk1) != AiPlaystyleFlags::none
+            && thoughtTypeHasFlags(thought.type, ThoughtTypeFlags::unk12))
+        {
+            return true;
+        }
+
+        if ((company.aiPlaystyleFlags & AiPlaystyleFlags::unk2) != AiPlaystyleFlags::none
+            && thoughtTypeHasFlags(thought.type, ThoughtTypeFlags::unk13))
+        {
+            return true;
+        }
+
+        if ((company.aiPlaystyleFlags & AiPlaystyleFlags::unk3) != AiPlaystyleFlags::none
+            && thoughtTypeHasFlags(thought.type, ThoughtTypeFlags::unk14))
+        {
+            return true;
+        }
+
+        if ((company.aiPlaystyleFlags & AiPlaystyleFlags::noAir) != AiPlaystyleFlags::none
+            && thoughtTypeHasFlags(thought.type, ThoughtTypeFlags::airBased))
+        {
+            return true;
+        }
+
+        if ((company.aiPlaystyleFlags & AiPlaystyleFlags::noWater) != AiPlaystyleFlags::none
+            && thoughtTypeHasFlags(thought.type, ThoughtTypeFlags::waterBased))
+        {
+            return true;
+        }
+
+        if ((company.aiPlaystyleFlags & AiPlaystyleFlags::unk7) != AiPlaystyleFlags::none
+            && thoughtTypeHasFlags(thought.type, ThoughtTypeFlags::unk7))
+        {
+            return true;
+        }
+
+        if ((company.aiPlaystyleFlags & AiPlaystyleFlags::unk6) != AiPlaystyleFlags::none
+            && !thoughtTypeHasFlags(thought.type, ThoughtTypeFlags::unk7))
+        {
+            return true;
+        }
+
+        if ((company.aiPlaystyleFlags & AiPlaystyleFlags::townIdSet) != AiPlaystyleFlags::none)
+        {
+            // The first thought of this play style must be located near to the home town chosen for the company
+            // this ensures that the company is based around that town
+            bool noOtherThoughts = true;
+            for (auto& otherThought : company.aiThoughts)
+            {
+                if (&otherThought == &thought)
+                {
+                    continue;
+                }
+                if (otherThought.type != AiThoughtType::null)
+                {
+                    noOtherThoughts = false;
+                    break;
+                }
+            }
+            if (noOtherThoughts)
+            {
+                auto destinationReferenceHomeTown = [&company](bool isIndustry, uint8_t destination) {
+                    if (isIndustry)
+                    {
+                        const auto* industry = IndustryManager::get(static_cast<IndustryId>(destination));
+                        const auto* homeTown = TownManager::get(static_cast<TownId>(company.aiPlaystyleTownId));
+                        const auto distance = Math::Vector::manhattanDistance2D(Pos2{ industry->x, industry->y }, Pos2{ homeTown->x, homeTown->y });
+                        return distance < 33 * 32;
```

Comment:
Is this 32 as in tile size?
```suggestion
                        return distance < 33 * kTileSize;
```